lib/Service/FileService.php

```diff
@@ -184,17 +189,25 @@ public function setDescendantDetection(Folder $folder, bool $detection): bool {
 	}
 
 	/**
-	 * Returns if the file is inside a shared storage.
+	 * Returns if the file is inside a native home storage.
+	 */
+	public function isUserFile(Node $node): bool {
+		return (($node->getMountPoint()->getMountType() === '') &&
+		        ($node->getStorage()->instanceOfStorage(IHomeStorage::class)));
+	}
+
+	/**
+	 * Returns if the file is inside a shared mount storage.
 	 */
 	public function isSharedFile(Node $node): bool {
-		return $node->getStorage()->instanceOfStorage(SharingExternalStorage::class);
+		return ($node->getMountPoint()->getMountType() === 'shared');
```

Comment:
this is proper way to detect shared/local? I mean, with mountType and not with `instanceOfStorage`? If it is...why I used `instanceOfStorage` 🤔 ? Did I just overlooked this one, or this was added later?

Reply:
What I discovered in recent tests, is that files shared by another user from the same instance are reported within HomeStorage. So,  it did not even run at this check. They were directly considered as local files. Therefore, to the local files I also verify that the mountype is also empty.

In my previous tests I only tested it with an external instance.. :disappointed:

Reply:
I didn't know about `mountType` when I wrote it. Maybe I should just have used this from start:) Good you fixed it:D

Reply:
MM.. I think it was a copypast made by me from another application.. :sweat_smile: 
But without enabling shared files (That still is experimental) everything worked fine. 

Well, I hope you can finally analyze your images!. :smile: